src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersTests.cs

```diff
@@ -421,27 +421,199 @@ struct ReallyLongName
 
     string S { get; }|]
 }",
+@"using System;
+using System.Collections.Generic;
+
+struct ReallyLongName : IEquatable<ReallyLongName>
+{
+    int i;
+
+    string S { get; }
+
+    public override bool Equals(object obj)
+    {
+        return obj is ReallyLongName name && Equals(name);
```

Comment:
❔ Do we have a test for cases where `obj is ReallyLongName name` is not supported by the current language version?

Reply:
yup. `TestEqualsSingleField` does this by passing in C#6.  It validates you get:

```
    public override bool Equals(object obj)
    {
        var program = obj as Program;
        return program != null &&
               a == program.a;
    }
```